src/plugins/dashboard/public/attribute_service/attribute_service.tsx

```diff
@@ -124,7 +124,10 @@ export class AttributeService<
           newAttributes,
           savedObjectId
         );
-        return { ...originalInput, savedObjectId: savedItem.id } as RefType;
+        if ('id' in savedItem) {
+          return { ...originalInput, savedObjectId: savedItem.id } as RefType;
+        }
+        return { ...originalInput } as RefType;
```

Comment:
I'm wondering about this case for when there isn't an id in the returned saved item - is this part of the error case that you've added to the API? If so, shouldn't we return the error?

Reply:
The `id` check here is to make typescript happy. But good point - I will add error handling here.

Reply:
Okay yes, makes sense. You need the check because you changed the signature of the custom save method to send back an error case.